pkg/controller/podautoscaler/horizontal.go

```diff
@@ -453,7 +453,12 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 		case desiredReplicas == 0:
 			//  never scale down to 0, reserved for disabling autoscaling
 			setCondition(hpa, autoscalingv2.ScalingLimited, v1.ConditionTrue, "TooFewReplicas", "the desired replica count was zero")
-			desiredReplicas = 1
+			// make sure we aren't below our minimum if one is defined
+			if hpa.Spec.MinReplicas != nil {
+				desiredReplicas = *hpa.Spec.MinReplicas
```

Comment:
Sorry, I mean both adding a warning message and setting `desiredReplicas` as your current PR here, make sense?